embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/DataCaptureServiceOtelConverter.kt

```diff
@@ -0,0 +1,22 @@
+package io.embrace.android.embracesdk.arch
+
+import io.embrace.android.embracesdk.internal.payload.Span
+
+/**
+ * Converts the data captured by a [DataCaptureService] into OpenTelemetry spans. This approach
+ * can be desirable as opposed to using a [DataSource] when Embrace has requirements that are
+ * incompatible with OTel's restrictions.
+ *
+ * For example, if we want to limit SpanEvents so that only the last N are captured, it's
+ * non-trivial to attempt that with the OTel SDK. Instead, we should use this interface to
+ * convert existing services to OTel telemetry, and then add it directly to our payloads.
+ *
+ * This has the disadvantage of the data is not sent in exporters, so should be used sparingly.
+ */
+internal fun interface DataCaptureServiceOtelConverter {
+
+    /**
+     * Returns a snapshot of the data captured by the service as a list of spans.
+     */
+    fun snapshot(): List<Span>?
```

Comment:
The way I'm thinking about this right now, `snapshot()` will effectively get called instead of `getCapturedData()` by the `PayloadMessageCollator`. This data won't get passed to an exporter & will only be added to the payload that gets sent to Embrace.

I'll merge as-is right now & can rename in a later changeset if you think there's a better name, or if you think the function shouldn't work as I've described